org.eclipse.jdt.ls.core/schema/org.eclipse.jdt.ls.core.codeLensProviderContainer.exsd

```diff
@@ -0,0 +1,112 @@
+<?xml version='1.0' encoding='UTF-8'?>

```

Comment:
We have configurations to disable code lenses for a particular codelens provider and a separate one for all codelenses. Which works in conjunction with dynamic capabilities to provide some optimization. I think this extension point needs to be enhanced to support those cases. Otherwise we are going to lose some of the optimizations around code lenses.

Reply:
I considered this , and that is why I included  `PreferenceManager` as a parameter of the interface method `CodeLensProvider getCodeLensProvider(String providerId, PreferenceManager pm)`. But currently class `Preferences` isn't that agile, namely if we want to add more configuration for new codelens, we have to update `Preferences` class to add new fields. I'm not sure that's what we want if we want to move codelensProvider to extensions, so I haven't updated it yet.

Reply:
Actually the Preferences instance is sufficient for all codelensproviders, as it exposes the raw preferences map sent by the client, via `preferences.asMap()`. That allows 3rd party adopters to use the java.* namespace for their preferences, without jdt.ls knowing anything about them (see [jdt.ls.decompiler](https://github.com/dgileadi/dg.jdt.ls.decompiler/blob/master/dg.jdt.ls.decompiler.fernflower/src/dg/jdt/ls/decompiler/fernflower/FernflowerDecompiler.java#L52)).

Reply:
cool, let me check about that